src/project.coffee

```diff
@@ -39,6 +39,20 @@ class Project extends Model
     @emitter = new Emitter
     @buffers ?= []
 
+    # Mapping from the real path of a {Directory} to a {Promise} that resolves
+    # to either a {Repository} or null. Ideally, the {Directory} would be used
+    # as the key; however, there can be multiple {Directory} objects created for
+    # the same real path, so it is not a good key.
```

Comment:
Aren't the keys in a `Map` compared via pointer equality? If instantiate two identical `Directory` objects, and store the `Repository` for one of them in this map, will I be able to read the repository out using the other?